### ISSUE TITLE
SMS messages not working: "datacoding must be plain, unicode or auto","parameter":"datacoding"

### ISSUE THREAD

**Issue by fejanto** [messagebird/java-rest-api#221]
Hi everybody.

I'm trying to send a simple SMS using 

`messageBirdClient.sendMessage("Sender", "Test Body", phones).getId();`

but i'm getting  "datacoding must be plain, unicode or auto","parameter":"datacoding", so I can't get it working since datacoding is set internally.

It míght be a problem on MessageBirdServiceImpl line 606, the body gets serialized like this:

`{"originator":"Sender","body":"Test Body","recipients":"541111111111","datacoding":"DataCoding{value='plain'}"}`

In summary, SMS messages are not working.

**Comment by denizkilic** [comment]
Hi @fejanto,
Thanks for creating the issue. I did a test using https://github.com/messagebird/java-rest-api/blob/master/examples/src/main/java/ExampleSendMessage.java and SMS message is sent successfully. ```
```Sending message:
MessageResponse{id='f355390647ad47eea0573d47e718fd26', href='https://rest.messagebird.com/messages/f355390647ad47eea0573d47e718fd26', direction='mt', type=MsgType{value='sms'}, originator='+316*******', body='testdeniz', reference='null', validity=null, gateway=10, typeDetails={}, datacoding=DataCoding{value='plain'}, mclass=MClass{value=1}, scheduledDatetime=null, createdDatetime=Thu Dec 01 17:30:14 CET 2022, recipients=Recipients{totalCount=1, totalSentCount=1, totalDeliveredCount=0, totalDeliveryFailedCount=0, items=[Items{recipient=316********, originator=null, status='sent', statusDatetime=Thu Dec 01 17:30:14 CET 2022, recipientCountry='null', recipientCountryPrefix=null, recipientOperator='null', messageLength=null, statusReason='null', price=null, mccmnc='null', mcc='null', mnc='null', messagePartCount=1}]}}
```
```
Could you please share your test code or more details about the test?

**Comment by ravi-signal** [comment]
@denizkilic Having the same problem with the Verify api, see #220 . I'm not sure how you're avoiding it, any serialized object with `DataCoding` set (which is set by default to `DataCoding.plain`) should cause the issue (introduced in [5.1.1](https://github.com/messagebird/java-rest-api/releases/tag/v5.1.1))

I can work around it by `setDatacoding(null)`, which uses the default encoding `plain` (but then I can't actually set the datacoding)

**Comment by fejanto** [comment]
Hi @denizkilic, thanks for the reply. I literally copy pasted the code in that example into my project and it doesn't work.

@ravi-signal in my case, i went back to v4.0.0 and the same code works perfect. Maybe that works for you too.

**Comment by fejanto** [comment]
@denizkilic I'm not sure I can write a test to verify this, because the error is not in the sdk itself. The failure is in the api call because the endpoint "doesn't like" the value sent in `datacoding` field.

However, as I said in last comment, if I switch from v5.1.2 to 4.0.0, it works just fine.

**Comment by denizkilic** [comment]
Hi @fejanto, I released a new version [5.3.0](https://github.com/messagebird/java-rest-api/releases/tag/v5.3.0) this should solve the problem. Can you try the latest release?
Hi @ravi-signal, thanks for pointing out the problem. I updated the Language class as well for the Verify API problem, could you please try the latest release?

**Comment by fejanto** [comment]
@denizkilic [5.3.0](https://github.com/messagebird/java-rest-api/releases/tag/v5.3.0) works like a charm. Thanks!

**Comment by ravi-signal** [comment]
@denizkilic same here, 5.3.0 fixes the issue with verify (and the language field still works)
